actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -1007,8 +1010,8 @@ def extra_tags_for_form(html_options)
 
         def form_tag_html(html_options)
           extra_tags = extra_tags_for_form(html_options)
-          html = tag(:form, html_options, true) + extra_tags
-          prevent_content_exfiltration(html)
+          html = content_tag(:form, extra_tags, html_options).then { |html| html.delete_suffix!("</form>") }
```

Comment:
Building a tag, just to remove the closing tag, and then calling `html_safe` doesn't seem worthy doing. You are making the framework do more work than it is necessary. This has performance impact.

Reply:
This change would be unnecessary if https://github.com/rails/rails/pull/44275/ were merged.